### ISSUE TITLE
Indexing blocked on named pipe

### ISSUE THREAD

**Issue by filippog** [anchore/syft#568]
**What happened**:
I'd like to use `syft` to catalog packages for all containers running on a Linux host and I'm using podman. I tried a `dir` indexing of the container's `MergedDir` like the following however indexing blocks on reading `/run/initctl`:

```
# syft -vv packages dir:$(podman inspect -f "{{ .GraphDriver.Data.MergedDir }}" e0b0c7861900)
...
[0001]  INFO indexing filesystem path="/run/lock"
[0001]  INFO indexing filesystem path="/run"
```

And strace shows

```
[pid 1030638] futex(0xc0139add50, FUTEX_WAIT_PRIVATE, 0, NULL <unfinished ...>
[pid 1030637] futex(0xc0139ac150, FUTEX_WAIT_PRIVATE, 0, NULL <unfinished ...>
[pid 1030634] futex(0x14460f8, FUTEX_WAIT_PRIVATE, 0, NULL <unfinished ...>
[pid 1030633] futex(0xc000052950, FUTEX_WAIT_PRIVATE, 0, NULL <unfinished ...>
[pid 1030632] epoll_pwait(4,  <unfinished ...>
[pid 1030631] restart_syscall(<... resuming interrupted read ...> <unfinished ...>
[pid 1030630] futex(0x14461a0, FUTEX_WAIT_PRIVATE, 0, NULL <unfinished ...>
[pid 1030632] <... epoll_pwait resumed>[], 128, 0, NULL, 0) = 0
[pid 1030632] epoll_pwait(4, ^Cstrace: Process 1030630 detached
```

lsof

```
root@host1:~# lsof -p 1030630
COMMAND     PID USER   FD      TYPE   DEVICE SIZE/OFF     NODE NAME
syft    1030630 root  cwd       DIR    254,1     4096   655374 /root
syft    1030630 root  rtd       DIR    254,1     4096        2 /
syft    1030630 root  txt       REG    254,1 16850944   655556 /root/syft
syft    1030630 root    0u      CHR    136,1      0t0        4 /dev/pts/1
syft    1030630 root    1u      CHR    136,1      0t0        4 /dev/pts/1
syft    1030630 root    2u      CHR    136,1      0t0        4 /dev/pts/1
syft    1030630 root    3u     IPv4 22790074      0t0      TCP host1:57828->172.67.71.75:https (ESTABLISHED)
syft    1030630 root    4u  a_inode     0,13        0     8042 [eventpoll]
syft    1030630 root    5r     FIFO     0,12      0t0 22790065 pipe
syft    1030630 root    6w     FIFO     0,12      0t0 22790065 pipe
syft    1030630 root    7r     FIFO     0,23      0t0      208 /run/initctl
```

**What you expected to happen**:

Regular files are scanned while sockets, named pipes, etc are skipped

**How to reproduce it (as minimally and precisely as possible)**:

Attempt `dir` scanning of directories with special files

**Anything else we need to know?**:

I don't know if `dir` + `podman inspect` is the correct way to index a running container

**Comment by dakaneye** [comment]
Hi @filippog thanks for filing this issue. I'd like to ask for a bit more clarifying information so I can assist you more effectively.

Am I correct in saying that you're wanting syft to report the named pipes and sockets in addition to the files that are in the container? 

Also it'd be helpful if you showed the syft output from the command (feel free to truncate) and show me exactly what you'd expect to be in that output. 

@wagoodman I'm not sure if this is a feature request or a bug to be honest, were the things that @filippog mentioned originally supposed to be catalogued? I always understood it to be scanning directories for files and packages specifically.

**Comment by luhring** [comment]
>Am I correct in saying that you're wanting syft to report the named pipes and sockets in addition to the files that are in the container?

My read on this is that Syft has historically indexed files of all types (including FIFOs, etc.), and this is undesirable.

>I always understood it to be scanning directories for files and packages specifically.

Yes exactly — this is the behavior we intend for local filesystem scans.

---

I'm wondering if #616 was accidentally a **duplicate** of this issue (caused by my failing memory regarding this issue! 🙈 ).

#616 has been fixed ✅. We should confirm that we can still reproduce this, and close this issue if we can't.

**Comment by dakaneye** [comment]
@filippog ok! based on @luhring 's comment can you send me some detailed steps to reproduce? I see in your original report you're using a container ID, and I would like to see if I can get the image name (if generic) that you're using. 

Likewise, @wagoodman if you have steps to follow for reproducing #615 that would be great too. Thanks!

**Comment by filippog** [comment]
Thank you for the assistance @dakaneye and @luhring ! I'm pretty sure this has been fixed in #615 as the issue is exactly the same (i.e. scanning non-regular files and blocking while doing so). Resolving!
